### ISSUE TITLE
a "maybe you forgot to reload" hint would do wonders

### ISSUE THREAD

**Issue by leipzig** [Supervisor/supervisor#582]
I can't be alone here having tried to restart and debug a conf file and then suddenly remembering it has to be reloaded in this particular tool.

Just a thought:
```
/usr/local/bin/supervisorctl start production
production: ERROR (no such file). Maybe you forgot to reload?
```

**Comment by msabramo** [comment]
Yeah you're not alone. The thing that happens with us more commonly and is tricky is that someone updates settings for an existing app and someone forgets to do supervisorctl update so it's still running with old settings. 

It would be neat if you could query for what settings are in use and/or the most recent last modified time (or a SHA-1?) for all of the conf files that were read at the time of the last read. The supervisorctl could warn when one of your conf files is newer than what is in effect and suggest that you do an update.

**Comment by msabramo** [comment]
I'd be willing to work on this if @mnabarez and @mcdonc gave some high-level direction.

**Comment by mnaberez** [comment]
> It would be neat if you could query for what settings are in use

The `reread` command will reread the config and show differences to what is on disk without making any changes to the active configuration.

**Comment by msabramo** [comment]
Oh, that's what `reread is`. :-)

Dumb as it sounds I never quite looked into what `reread` does. I guess I could stand to RTFM a little bit more.

**Comment by msabramo** [comment]
I wonder if the `reread` functionality could be incorporated into `status` and the Web UI?

I feel like a lot of folks including myself mainly stick to start, stop, and status and/or the Web UI. They'd be more likely to notice a stale config in those places.

**Comment by mnaberez** [comment]
Originally posted in #1552:

> I'd like to see if a process should be restarted when displaying the `status`.
> It's easy to forget to `reread` the configuration file after modifying it. In such case, the user try to `restart` it and does not understand why the changes are not used.
> 
> Currently, the output is:
> ```
> supervisor> status
> demo                             RUNNING   pid 16371, uptime 0:02:43
> demo2                            RUNNING   pid 16372, uptime 0:02:43
> ```
> 
> It could be something like:
> ```
> supervisor> status
> demo                             RUNNING  changed pid 16371, uptime 0:02:43
> demo2                            RUNNING  -       pid 16372, uptime 0:02:43
> ```
> 
> Adding such info would break the backward compatibility if scripts parse the output.
> To address this new problem, a configuration could be added to set the output of `status` in `/etc/supervisor/supervisord.conf` file. I think about two ways to implement it:
> 
> - 1. a boolean (called like `changedstatus`) to insert/ignore this data. By default, it must be disabled (for backward compatibility). 
> - 2. a list (called like `statusline`) to set up data displayed and their order. For example, with `name,pid,status`, a line will be like:
> ```
> demo                             pid 16371, uptime 0:02:43  RUNNING
> ```
> 
>    By default, it would be equivalent to `statusline=name,status,pid` (for backward compatibility).
> 
> IMO, Solution 2 is better because it's more versatile and allows to add new data in the future if needed.
> 
> Another way would be to add a new command to do that but the general usage of `supervisorctl` will be less clear so I think it's solution 2 is better than this one too.
> 
> What do you think about:
> - the idea to add the `changed` data in `status` command?
> - the way to implement backward compatibility? 
> 
> I'm interested to make the PRs to implement it.
